compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -14117,6 +14117,168 @@ bool isSettingOp(uint8_t byteValue, TR::InstOpCode::Mnemonic SI_opcode)
+            TR_ASSERT(0, " Unsupported DataType for bitPermute op");
+            }
+         if (node->getDataType() != TR::Int64)
+            {
+            // Java Spec defines the shift amount for integer shifts to be masked with
+            // 0x1f (0-31). For long shifts, the shift instruction will only consider
+            // the first 6 bits of the operand as the shift amount. So no mask is required in that case
+            generateRIInstruction(cg, TR::InstOpCode::NILL, node, tmpReg, 0x1f);   
+            //TODO: There should be two of these
+            }
+         //create memory reference using tmpReg (shift amount), then shift valueReg by by shift amount
+         // TODO: Should another register be used here instead of tmpReg?
+         TR::MemoryReference * tempMR = generateS390MemoryReference(tmpReg, 0, cg);
+         generateRSInstruction(cg, TR::InstOpCode::getShiftRightLogicalSingleOpCode(), node, tmpReg, valueReg, tempMR); 
+         tempMR->stopUsingMemRefRegister(cg); 
+
+         if (node->getDataType() != TR::Int64)
+            {
+            // now and the value in tmpReg by 1 to get your relevant bit
+            generateRILInstruction(cg, TR::InstOpCode::NILF, node,tmpReg,0x1);
+            }
+         else
+            {
+            // This is equivalent to doing a `tmpReg & 0x1` (There should be a helper for this to check processor info)
+            TR::InstOpCode::Mnemonic opCode = cg->getS390ProcessorInfo()->supportsArch(TR_S390ProcessorInfo::TR_zEC12) ? TR::InstOpCode::RISBGN : TR::InstOpCode::RISBG;
+            generateRIEInstruction(cg, opCode, node, tmpReg, tmpReg, 63, 191, 0);
+            }
+         //now left shift by x to get to the position u want
+         if (x > 0)
+            generateRSInstruction(cg, TR::InstOpCode::getShiftLeftLogicalSingleOpCode(), node, tmpReg, tmpReg, x);
+
+         // now OR the result into the resultReg
+         generateRRInstruction(cg, TR::InstOpCode::getOrRegOpCode(), node, resultReg, tmpReg);
+         }
+      }
+   else
+      {
+      auto lengthReg = cg->evaluate(length);
+      auto indexReg = cg->allocateRegister(TR_GPR);
+
+      TR::RegisterDependencyConditions* deps = generateRegisterDependencyConditions((uint8_t)0, (uint8_t)4, cg);
+      deps->addPostCondition(addrReg, TR::RealRegister::AssignAny);
+      deps->addPostCondition(indexReg, TR::RealRegister::AssignAny);
+      deps->addPostCondition(valueReg, TR::RealRegister::AssignAny);
+      deps->addPostCondition(tmpReg, TR::RealRegister::AssignAny);
+
+      TR::LabelSymbol *startLabel = generateLabelSymbol(cg);
+      TR::LabelSymbol *endLabel = generateLabelSymbol(cg);
+      startLabel->setStartInternalControlFlow();
+      endLabel->setEndInternalControlFlow();
+
+      // Load array length into index and test to see if it's already zero
+      // cc=0 if register value is 0, cc=1 if value < 1, cc=2 if value>0
+      generateRRInstruction(cg, TR::InstOpCode::LTGR, node, indexReg, lengthReg); 
```

Comment:
Do we need to be careful here if running on 31-bit? What is the type of `lengthReg` on 31-bit?

Reply:
Actually I think we can get away with using an `LTR` here because lengthReg is always a `TR::Int32` as per the IL.

Reply:
Just to confirm, the correct implementation here would be:
`generateRRInstruction(cg, TR::InstOpCode::getLoadTestRegOpCode(), node, indexReg, lengthReg);`

where the `getLoadTestRegOpCode()` query is defined as below:
https://github.com/eclipse/omr/blob/58194bc93b1569ed1fe2a994fb0b0d7276a9e0e2/compiler/z/codegen/InstOpCode.cpp#L254-L255

Reply:
This was done incorrectly before. The correct query was added here: f1f0697